tests/client/test_search.py

```diff
@@ -58,6 +58,19 @@ def test_field_regex_criteria():
     ) == {"some.field": {"$regex": "abc"}}
 
 
+def test_field_less_than_criteria():
+    """with_filed with less_than is translated as expected for
+    date and non-date types
+    """
+    c1 = Criteria.with_field("num_field", Matcher.less_than("5"))
+    c2 = Criteria.with_field("date_field", Matcher.less_than("2019-08-27T00:00:00Z"))
```

Comment:
This spot again would need changing to support passing a datetime.datetime.

I think this test shows the inconsistency of current setup - when caller passes less_than("5"), the client doesn't guess that 5 should be interpreted as an int.  But when caller passes less_than("2019-08-27T00:00:00Z"), it does guess that it should be interpreted as a date.

Reply:
Agree. Should we consider int type fields for a similar conversion as well?(as a separate PR)

Reply:
Hmm... I'm not really sure... I guess it's not as useful as the datetime conversion at least.

There's a reasonable argument for str => datetime conversion in that there is no supported datetime type across languages, and an ISO8601 string is really the de-facto standard for datetime interchange.  If a datetime is coming from a user, it probably was provided as an ISO8601 string in the first place.

For str => int, I guess it's not as useful, seems like it just enables the caller to be sloppy with their types, which could be a bad thing.

Needs more thought.  Either way, as you said let's not try to handle it in this PR.